library/src/blas1/rocblas_swap.cpp

```diff
@@ -48,7 +48,7 @@ namespace
 
         RETURN_ZERO_DEVICE_MEMORY_SIZE_IF_QUERIED(handle);
 
-        return rocblas_swap_template<NB>(handle, n, x, incx, y, incy);
+        return rocblas_swap_template<NB, T>(handle, n, x, 0, incx, 0, y, 0, incy, 0, 1);
     }
```

Comment:
I like naming constant values:

static constexpr ptrdiff_t zero_offset =0;
rocblas_swap_template<NB, T>(handle, n, x, zero_offset , incx, zero_offset , y, zero_offset , incy, zero_offset , 1)

Reply:
I had some of those but to be consistent with other functions I removed them.   Intellisense can show you the parameter names so that helps a bit too.

Reply:
I prefer `batch_count` to have a default value of `1` in the function header, rather than passing 1 explicitly in the call.

`rocblas_swap_template<NB,T>(...)` can be simplified to `rocblas_swap_template<NB>(...)`. That's the reason for putting type template parameters at the end.

Reply:
https://github.com/ROCmSoftwarePlatform/rocBLAS/blob/b4a317d2f3777cd656770a85fad22745d5d10618/library/src/blas1/rocblas_swap.hpp#L48

```
         rocblas_int    batch_count = 1)
```
This documents the default `batch_count` pretty clearly without having to use named constants like `one_batch`, and it avoids passing `1` at the end of a call.